### ISSUE TITLE
Bug in AssignedGenerator.php

### ISSUE THREAD

**Issue by janelletavares** [doctrine/doctrine2#6091]
I believe that I found a bug in the latest version of AssignedGenerator.php.

- $value = $em->getUnitOfWork()->getSingleIdentifierValue($value);
+ $value = $em->getUnitOfWork()->getSingleIdentifierValue($entity);

getSingleIdentifierValue is expecting an entity.  When you pass a value into getSingleIdentifierValue and not an entity, the get_class() function does not return a coherent value. A type hint, if that is possible, would help. In my case, get_class() has been returning an empty string.   Eventually, the code throws a MappingException::nonExistingClass exception.

Here's generally what I was doing when I reproduced it.

class MyStuff {
  /*
   * @Id @Column(type="string", length=255, unique=true)
   */
  private $key;

  /*
   * @ManyToOne(targetEntity="YourStuff")
   *
   * @Id @Column(type="string", nullable=false, length=255)
   *
   * @JoinColumn(name="name_join", referencedColumnName="key")
   */
  private $foreign_key;
}

class YourStuff {
  /*
   * @Id @Column(type="string", length=255, unique=true)
   */
  private $key;
}

**Comment by Ocramius** [comment]
@janelletavares can you please try reproducing this in a test case? See https://github.com/doctrine/doctrine2/tree/d3f6c5ec70aac4b029a4b61ecf1e2ba61a1a4a6d/tests/Doctrine/Tests/ORM/Functional/Ticket for examples

**Comment by janelletavares** [comment]
I'm not sure when I will have time to do that, but I will keep it on my ToDo list.

**Comment by fezfez** [comment]
big todo list

**Comment by Ocramius** [comment]
@fezfez pretty much like everyone. Indeed, without a test we can't do much about this issue.

**Comment by fezfez** [comment]
@Ocramius : yeah i know, i was searching docs for AssignedGenerator and i found this issue. the reponse from @janelletavares has more than one year, i just laught about how big  his todo list must be.

No offense and thanks for your work ! ❤️
